### PR TITLE
Feat: dashboard details view 

### PR DESCRIPTION
**Pull request by PikkaPikkachu**
## Context

Build a detailed dashboard view of the batch letters for the new UI revamp, as in the figma [here](https://www.figma.com/file/KVFsGATLUfvPqKE2lY9niH/LetterSG?node-id=1634%3A55391&mode=dev). 

## Approach

1. Build new components for the detailed batch view page. 
2. We also remove the old components that not being used anymore. 
3. This is mostly a frontend change, with a bit of change in the dto objects for further compatibility and reducing code base's complexity. 

**Features**:

1. When users click on the batch, they are redirected to the detailed batch letters page. 
2. They are shows certain stats about the batch. 
3. Officers can see the individuals letters that they issued and click on any of the rows to open a drawer for more information on the letters. 

## Before & After Screenshots

**AFTER**:

https://github.com/opengovsg/letters/assets/25703976/047e1332-c815-485c-9f5c-46d3bfef67f3

### CODE REVIEW

**Comment by gweiying** [comment]
Holding off on merge until next release, so will get to reviews probably later this week

File: backend/src/database/entities/batch.entity.ts [hunk 1]
```diff
@@ -32,6 +32,14 @@ export class Batch {
   @Column({ type: 'text', default: '' })
   passwordInstructions: string
 
+  @Column({ type: 'boolean', default: false })
+  isPasswordProtected: boolean
```
Comment (gweiying) [inline backend/src/database/entities/batch.entity.ts:36]: side comment, will probably need a strategy on how to backfill this for existing batches!

File: backend/src/database/entities/batch.entity.ts [hunk 2]
```diff
@@ -32,6 +32,14 @@ export class Batch {
...
+
+  // channel is not enforced as an enum on DB layer
+  // allows us to avoid future migrations if we extend to other channels
+  @Column('text', { nullable: true })
+  notificationChannel: string
```
Comment (gweiying) [inline backend/src/database/entities/batch.entity.ts:41]: side comment, will probably need a strategy on how to backfill this for existing batches!

**Comment by PikkaPikkachu** [comment]
This is getting a bit hard to rebase at this point, will open a new PR and follow up on the comments there :)

**Pull request closed without merging**